test/bash/util.bats

```diff
@@ -0,0 +1,159 @@
+
+setup() {
+  load '../../node_modules/bats-support/load'
+  load '../../node_modules/bats-assert/load'
+
+  TEST_DIR="$( cd "$( dirname "$BATS_TEST_FILENAME" )" >/dev/null 2>&1 && pwd )"
+
+  # Not used yet, but will be later
+  TEST_ROOT_DIR=$(mktemp -u)
+
+  load "$TEST_DIR/../../src/usr/local/buildpack/util.sh"
+}
+
+teardown() {
+  rm -rf "${TEST_ROOT_DIR}"
+}
+
+@test "require_tool" {
+  run require_tool
+  assert_output --partial "No tool defined"
+  assert_failure
+
+  # TODO(Chumper): In my opinion this test should fail
+  run require_tool foo
+  # assert_output --partial "No version defined"
+  # assert_failure
+  assert_success
+
+  require_tool foo 1.2.3
+  assert [ "${TOOL_NAME}" = "foo" ]
+  assert [ "${TOOL_VERSION}" = "1.2.3" ]
+  assert [ "${FOO_VERSION}" = "1.2.3" ]
+
+  FOO_VERSION=1.2.4 require_tool foo
+  assert [ "${TOOL_NAME}" = "foo" ]
+  assert [ "${TOOL_VERSION}" = "1.2.4" ]
+
+  FOO_BAR_VERSION=1.2.5 require_tool foo-bar
+  assert [ "${TOOL_NAME}" = "foo-bar" ]
+  assert [ "${TOOL_VERSION}" = "1.2.5" ]
+}
+
+@test "requires_user" {
+  run require_user
+  assert_output --partial "No USER_NAME defined"
+  assert_failure
+
+  # TODO(Chumper): In my opinion this test should fail
+  USER_NAME= run require_user
+  # assert_output --partial "No USER_NAME defined"
+  # assert_failure
+  assert_success
+
+  USER_NAME=foo run require_user
+  assert_success
+}
+
+@test "requires_root" {
+  run require_root
+
+  # Bogus test right now, but will be improved
+  # when we refactor the utils functions
+  if [[ $EUID -ne 0 ]]; then
+    assert_output --partial "run as root"
+    assert_failure
+  else
+    assert_success
+  fi
+
+  USER_NAME=foo run require_user
+  assert_success
+}
+
+@test "check_semver" {
+
+  run check_semver
+  assert_failure
+
+  run check_semver foo.bar.baz
+  assert_failure
+
+  check_semver 1.2.3
```

Comment:
Test `11.0.14+9` too, just to be safe

https://github.com/containerbase/buildpack/blob/b09200b22e93dd61140d155a6e3c9c215b6ad5cd/test/java/Dockerfile#L19